pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -670,55 +695,61 @@ func doInstall(ctx *zedmanagerContext,
 		log.Infof("Found StorageStatus URL %s safename %s\n",
 			ss.Name, safename)
 
-		vs := lookupVerifyImageStatusAny(ctx, safename,
+		is := lookupImageStatusForApp(ctx, config.UUIDandVersion.UUID,
 			ss.ImageSha256)
-		if vs == nil || vs.Expired {
-			log.Infof("VerifyImageStatus for %s sha %s not found (%v)\n",
-				safename, ss.ImageSha256, vs)
-			// Keep at current state
-			minState = types.DOWNLOADED
-			changed = true
-			continue
-		}
-		if minState > vs.State {
-			minState = vs.State
-		}
-		if vs.State != ss.State {
-			ss.State = vs.State
-			changed = true
-		}
-		if vs.Pending() {
-			log.Infof("lookupVerifyImageStatusAny %s Pending\n",
-				safename)
-			continue
-		}
-		if vs.LastErr != "" {
-			log.Errorf("Received error from verifier for %s: %s\n",
-				safename, vs.LastErr)
-			ss.Error = vs.LastErr
-			ss.ErrorSource = pubsub.TypeToName(types.VerifyImageStatus{})
-			errorSource = ss.ErrorSource
-			allErrors = appendError(allErrors, "verifier",
-				vs.LastErr)
-			ss.ErrorTime = vs.LastErrTime
-			errorTime = vs.LastErrTime
-			changed = true
-			continue
-		} else if ss.ErrorSource == pubsub.TypeToName(types.VerifyImageStatus{}) {
-			log.Infof("Clearing verifier error %s\n", ss.Error)
-			ss.Error = ""
-			ss.ErrorSource = ""
-			ss.ErrorTime = time.Time{}
-			changed = true
-		}
-		switch vs.State {
-		case types.INITIAL:
-			// Nothing to do
-		default:
-			ss.ActiveFileLocation = types.VerifiedAppImgDirname + "/" + vs.Safename
+		if is.AppInstUUID == config.UUIDandVersion.UUID {
```

Comment:
See comment about error/not-found checking.

Reply:
Done.. fixed.